### PR TITLE
feat: Height percentiles report (#719)

### PR DESCRIPTION
**Pull request by Lastpixl**
Hi,

I really like the new WHO **w**eight percentile report... So, here is my contribution proposal: a similar **h**eight percentile report (#719).

The implementation is really similar to that of the weight percentile report, it works in a dev environment. The instructions are great, writing this was pleasant! Kudos to the maintainers.
`makemessages` has generated fuzzy entries for french, I took the liberty of also fixing the few instances unrelated to this change for french.

Thanks for considering it for merging. Feel free to modify it in any way you see fit.

### CODE REVIEW

File: reports/graphs/height_change.py [hunk 1]
```diff
@@ -1,34 +1,107 @@
...
+    :param actual_heights: a QuerySet of Height instances.
+    :param percentile_heights: a QuerySet of Height Percentile instances.
+    :param birthday: a datetime of the child's birthday
     :returns: a tuple of the the graph's html and javascript.
     """
-    objects = objects.order_by("-date")
+    actual_heights = actual_heights.order_by("-date")
 
-    trace = go.Scatter(
+    weighing_dates: list[datetime] = list(actual_heights.values_list("date", flat=True))
```
Comment (cdubz) [inline reports/graphs/height_change.py:24]:
```suggestion
    measuring_dates: list[datetime] = list(actual_heights.values_list("date", flat=True))
```

File: reports/graphs/height_change.py [hunk 2]
```diff
@@ -1,34 +1,107 @@
...
+    measured_heights = list(actual_heights.values_list("height", flat=True))
+
+    actual_heights_trace = go.Scatter(
         name=_("Height"),
-        x=list(objects.values_list("date", flat=True)),
-        y=list(objects.values_list("height", flat=True)),
+        x=weighing_dates,
```
Comment (cdubz) [inline reports/graphs/height_change.py:29]:
```suggestion
        x=measuring_dates,
```

**Review by cdubz — CHANGES_REQUESTED** [review]
Thanks for contributing! Happy to hear it was easy to get going.

I have just one small CR for a variable naming.

Also could you modify: https://github.com/babybuddy/babybuddy/blob/1ef0bb9287adc4b47e74ec9cb2ac5164f142cc11/babybuddy/management/commands/fake.py#L89 to pick a starting fake height value that fits the model better? Right now when generating fake data and looking at these new reports we don't see any of the WHO lines because the test data is so much lower. Maybe set this to `self.height = round(uniform(45.0, 60.0), 2)`?

File: reports/graphs/height_change.py (hunk 1 above)
Comment (Lastpixl) [reply reports/graphs/height_change.py]: done

File: reports/graphs/height_change.py (hunk 2 above)
Comment (Lastpixl) [reply reports/graphs/height_change.py]: done

File: reports/graphs/height_change.py [hunk 3]
```diff
@@ -1,34 +1,107 @@
...
+        dates = list(
+            map(
+                lambda timedelta: birthday + timedelta,
+                percentile_heights.values_list("age_in_days", flat=True),
+            )
+        )
+
+        # reduce percentile data xrange to end 1 day after last weigh in for formatting purposes
+        # https://github.com/babybuddy/babybuddy/pull/708#discussion_r1332335789
+        last_date_for_percentiles = max(weighing_dates) + timedelta(days=2)
```
[the remark this answers is not in the document]
Comment (Lastpixl) [reply reports/graphs/height_change.py]: done

File: reports/graphs/height_change.py [hunk 4]
```diff
@@ -1,34 +1,107 @@
...
     layout_args["barmode"] = "stack"
     layout_args["title"] = _("<b>Height</b>")
     layout_args["xaxis"]["title"] = _("Date")
     layout_args["xaxis"]["rangeselector"] = utils.rangeselector_date()
     layout_args["yaxis"]["title"] = _("Height")
+    if percentile_heights:
+        # zoom in on the relevant dates
+        layout_args["xaxis"]["range"] = [
+            birthday,
+            max(weighing_dates) + timedelta(days=1),
```
[the remark this answers is not in the document]
Comment (Lastpixl) [reply reports/graphs/height_change.py]: done

File: reports/graphs/height_change.py [hunk 5]
```diff
@@ -1,34 +1,107 @@
...
 
+    if percentile_heights:
+        dates = list(
+            map(
+                lambda timedelta: birthday + timedelta,
+                percentile_heights.values_list("age_in_days", flat=True),
+            )
+        )
+
+        # reduce percentile data xrange to end 1 day after last weigh in for formatting purposes
```
[the remark this answers is not in the document]
Comment (Lastpixl) [reply reports/graphs/height_change.py]: done

**Comment by Lastpixl** [comment]
Thanks for the review, that was a lot of copypasta mistakes on my part... I have applied the requested changes, fixed an additional bug in my last commit.

**Comment by Lastpixl** [comment]
Should I rebase this against master and update translation files to fix the conflict with `locale/zh/LC_MESSAGES/django.po`?

**Comment by cdubz** [comment]
> Should I rebase this against master and update translation files to fix the conflict with `locale/zh/LC_MESSAGES/django.po`?

Oh shoot yeah I forgot this PR also had locale changes. Yes you’ll need to merge in or rebase on `master`.

**Comment by Lastpixl** [comment]
Rebased on master, re-ran `gulp makemessages -a` then `gulp compilemessages` as part of the rebase of the first commit, the one that adds new strings.

**Comment by Lastpixl** [comment]
One more rebase to fix formatting

**Comment by cdubz** [comment]
Looks good. Thanks for contributing! :santa:

**Pull request merged**